### PR TITLE
Move Psi/Phi curve generation out of Python loop

### PR DESCRIPTION
**Pull request by jeremykubica**
Move the generation of the psi and phi curves from inside a Python loop to inside a C++ loop. This is a very small speedup, but also adds very little complexity.

This PR also:

-  breaks up some of the giant `test_search.py` by moving a few tests to `test_stack_search_results.py`
-  Applies clang formatting to the C++ files.

### CODE REVIEW

File: src/kbmod/search/stack_search.cpp [hunk 1]
```diff
@@ -256,17 +256,33 @@ std::vector<float> StackSearch::extract_psi_or_phi_curve(Trajectory& trj, bool e
     return result;
 }
 
+std::vector<std::vector<float> > StackSearch::get_psi_curves(std::vector<Trajectory>& trajectories) {
```
Comment (gitosaurus) [inline src/kbmod/search/stack_search.cpp:259]: Any reason why the input isn't `const &`?

Comment (jeremykubica) [reply src/kbmod/search/stack_search.cpp]: No particularly good reason. I made that change (and also updated some of the other functions).

File: tests/test_stack_search_results.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        self.assertEqual(psi_curves.shape[1], self.num_times)
+        self.assertTrue(np.all(psi_curves > 0.0))
+
+        phi_curves = np.array(self.search.get_phi_curves(self.fake_trjs))
+        self.assertEqual(phi_curves.shape[0], self.num_objs)
+        self.assertEqual(phi_curves.shape[1], self.num_times)
+        self.assertTrue(np.all(phi_curves > 0.0))
+
+        # Check that the batch getters give the same results as the iterative ones.
+        for i in range(self.num_objs):
```
Comment (drewoldag) [inline tests/test_stack_search_results.py:70]: Nice, looks good.

**Review by drewoldag — APPROVED** [review]
This all looks pretty reasonable to me.

**Pull request merged**